zebrad/src/components/mempool/storage.rs

```diff
@@ -116,12 +127,15 @@ pub struct Storage {
     /// The set of verified transactions in the mempool.
     verified: VerifiedSet,
 
+    /// The set of outpoints with pending requests for their associated transparent::Output.
+    pub(super) pending_outputs: PendingOutputs,
+
     /// The set of transactions rejected due to bad authorizations, or for other
     /// reasons, and their rejection reasons. These rejections only apply to the
     /// current tip.
     ///
     /// Only transactions with the exact [`UnminedTxId`] are invalid.
-    tip_rejected_exact: HashMap<UnminedTxId, ExactTipRejectionError>,
+    tip_rejected_exact: HashMap<transaction::Hash, ExactTipRejectionError>,
```

Comment:
> What is the reason for this change?

I think the broader change to mined ids was to avoid iterating through the verified set to find the auth digest when adding transaction dependencies (since `OutPoint` only includes the mined id).

The change to `tip_rejected_exact` was a mistake, it would work fine for the `sendrawtransaction` RPC, but I see it would incorrectly reject gossiped transaction ids when it should try to download and verify them. 

Restored its type to the full `UnminedTxId` in https://github.com/ZcashFoundation/zebra/pull/8857/commits/ce6d169a9a0a1848b1d50a9eb29b7710f48043da.